### PR TITLE
Developer guide examples for Spring 4 plugin

### PR DESCRIPTION
**Pull request by qualidafial**
Addresses #738

### CODE REVIEW

**Comment by qualidafial** [comment]
@stevenschlansker or @arteam, could you please review?

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -2072,9 +2072,114 @@ To use this module, add a Maven dependency:
...
+       xmlns:aop="http://www.springframework.org/schema/aop"
+       xmlns:tx="http://www.springframework.org/schema/tx"
+       xsi:schemaLocation="
+       http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.0.xsd
+       http://www.springframework.org/schema/tx http://www.springframework.org/schema/tx/spring-tx-2.0.xsd
+       http://www.springframework.org/schema/aop http://www.springframework.org/schema/aop/spring-aop-2.0.xsd">
+
+  <!--1-->
+  <bean id="derby"
+    class="org.apache.derby.jdbc.EmbeddedDataSource"
```
Comment (stevenschlansker) [inline docs/src/adoc/index.adoc]: Should we consider converting the Derby stuff to h2, to be consistent with the rest of the tests?

**Comment by qualidafial** [comment]
@stevenschlansker What do you think of renaming `JdbiFactoryBean.installPlugins` to `.autoInstallPlugins`?

**Comment by qualidafial** [comment]
@stevenschlansker ping. I'm holding off on merging until I hear back from you on my question.

**Comment by qualidafial** [comment]
@stevenschlansker ping?

**Comment by stevenschlansker** [comment]
Sorry I was out of town.  Sounds fine to me.

**Pull request merged**